packages/client/src/__tests__/integration/happy/blog-env-mongo/test.ts

```diff
@@ -0,0 +1,279 @@
+import { generateTestClient } from '../../../../utils/getTestClient'
+
+// @ts-ignore
+import type { PrismaClient } from './node_modules/@prisma/client'
+
+const describeIf = (condition: boolean) => (condition ? describe : describe.skip)
+
+describeIf(!process.env.TEST_SKIP_MONGODB)('blog-env-mongo', () => {
+  let prisma: PrismaClient // Generated Client instance
+  let PrismaHelpers: any = null
+  const requests: any[] = []
+
+  beforeAll(async () => {
+    await generateTestClient()
+    const { PrismaClient, Prisma } = require('./node_modules/@prisma/client')
+
+    PrismaHelpers = Prisma
+    prisma = new PrismaClient({
+      errorFormat: 'colorless',
+      __internal: {
+        measurePerformance: true,
+        hooks: {
+          beforeRequest: (r: any) => requests.push(r),
+        },
+      },
+      log: [
+        {
+          emit: 'event',
+          level: 'error',
+        },
+      ],
+    })
+  })
+
+  afterEach(async () => {
+    for (const key of Object.keys(prisma)) {
+      await prisma[key]?.['deleteMany']?.()
+    }
+    await prisma.$disconnect()
+  })
+
+  test('includes version in generated client', () => {
+    const { Prisma } = require('./node_modules/@prisma/client')
+
+    const { prismaVersion } = Prisma
+
+    expect(prismaVersion).not.toBeUndefined()
+    expect(prismaVersion.client).not.toBeUndefined()
+  })
+
+  test('does not leak connection strings in node_modules', () => {
+    // @ts-ignore
+    expect(prisma.internalDatasources).toBeUndefined()
+  })
+
+  test('invokes beforeRequest hook', async () => {
+    await prisma.user.findMany()
+    expect(requests.length).toBeGreaterThan(0)
+  })
+
+  test('can throw validation errors', async () => {
+    const {
+      Prisma: { PrismaClientValidationError },
+    } = require('./node_modules/@prisma/client')
+
+    try {
+      await prisma.post.create({
+        // @ts-ignore
+        data: {},
+      })
+
+      expect(false).toBe(true) // The line above needs to throw, so this should never be executed, but if it does (aka the line above did not throw, as expected), it will fail the test
+    } catch (e) {
+      expect(e).not.toBeUndefined()
+      expect(e).toBeInstanceOf(PrismaClientValidationError)
+    }
+  })
```

Comment:
```suggestion
  test('can throw validation errors', async () => {
    expect.assertions(2)

    const {
      Prisma: { PrismaClientValidationError },
    } = require('./node_modules/@prisma/client')

    try {
      await prisma.post.create({
        // @ts-ignore
        data: {},
      })
    } catch (e) {
      expect(e).not.toBeUndefined()
      expect(e).toBeInstanceOf(PrismaClientValidationError)
    }
  })
```

Does the same but in a better way I think